woocommerce/wc-admin
# Build: Update i18n generation to combine JS + PHP translations

Currently the process to build translations, `npm run i18n`, doesn't automatically extract the JS strings. This PR cleans up the translation tasks, and adds the `pot` file generation into the build script too.

**To test**

- You should have a folder `/languages/` with just a readme in it (any other files can be deleted)
- Run `npm run i18n`
- Open `/languages/wc-admin.php`, this has all the JS strings extracted into one php file
- Open `/languages/wc-admin.pot`, this should have _all_ strings from JS and PHP inside
- Alternately, you can test by running `npm run build`, which will also build the JS files themselves

You can also check that the translations are loaded with the smallest Spanish translation (I've only added translations for "Analytics", "Orders", "Products", and "`{{productLink}}%s{{/productLink}} reviewed by {{authorLink}}%s{{/authorLink}}`" also note I don't actually _speak_ Spanish)

- [Download translations](https://github.com/woocommerce/wc-admin/files/2416420/translations.zip)
- Extract this to `wp-content/languages/plugins/`
- Switch wp-admin into Spanish in Settings > General, then visit a WooCommerce page
- You should see the translations in the sidebar menu names
- Now you can go to Analíticas > Pedidos (Orders), and the breadcrumbs should be in Spanish `WooCommerce / Analíticas / Pedidos`
- You can also open the reviews tab to see that `interpolateComponents` is working:

![screen shot 2018-09-25 at 1 53 21 pm](https://user-images.githubusercontent.com/541093/46032725-64b04300-c0ca-11e8-945e-b171a042dead.png)
